librz/bin/format/elf/elf_symbols.c

```diff
@@ -364,6 +364,78 @@ static bool get_section_elf_symbols(ELFOBJ *bin, RzVector *result, RzBinElfSymbo
 	return true;
 }
 
+static bool get_gnu_debugdata_elf_symbols(ELFOBJ *bin, RzVector *result, RzBinElfSymbolFilter filter, HtUU *set) {
+	// Get symbols from .gnu_debugdata according to https://sourceware.org/gdb/onlinedocs/gdb/MiniDebugInfo.html
+	bool res = false;
+	const RzBinElfSection *gnu_debugdata = Elf_(rz_bin_elf_get_section_with_name)(bin, ".gnu_debugdata");
+	if (!gnu_debugdata) {
+		return false;
+	}
+
+	RzBuffer *data_buf = rz_buf_new_slice(bin->b, gnu_debugdata->offset, gnu_debugdata->size);
+	if (!data_buf) {
+		return false;
+	}
+
+	RzBuffer *dec_buf = rz_buf_new_empty(0);
+	if (!dec_buf) {
+		goto data_buf_err;
+	}
+
+	if (!rz_lzma_dec_buf(data_buf, dec_buf, 1 << 13, NULL)) {
+		goto dec_buf_err;
+	}
+
+	RzBinObjectLoadOptions obj_opts = {
+		.baseaddr = UT64_MAX,
+		.loadaddr = 0,
+		.elf_load_sections = true,
+	};
+	ELFOBJ *debug_data_bin = Elf_(rz_bin_elf_new_buf)(dec_buf, &obj_opts);
+	if (!debug_data_bin) {
+		goto dec_buf_err;
+	}
+
+	RzVector *debug_symbols = Elf_(rz_bin_elf_compute_symbols)(debug_data_bin, filter);
+	if (!debug_symbols) {
+		goto debug_data_err;
+	}
+
+	HtPP *name_set = ht_pp_new0();
+	if (!name_set) {
+		goto debug_symbols_err;
+	}
+
+	RzBinElfSymbol *sym;
+	rz_vector_foreach(result, sym) {
+		ht_pp_insert(name_set, sym->name, sym);
```

Comment:
this will fail if the name is already in the map.

Reply:
it's alright, isn't it? i just don't want to provide dupes of symbols when parsing .gnu_debugdata